py-polars/polars/_utils/construction/dataframe.py

```diff
@@ -1020,6 +1020,9 @@ def pandas_to_pydf(
 ) -> PyDataFrame:
     """Construct a PyDataFrame from a pandas DataFrame."""
     convert_index = include_index and not _pandas_has_default_index(data)
+    if convert_index and (data.index.name in data.columns):
+        msg = "cannot convert pandas DataFrame when index name duplicates some column name"
+        raise ValueError(msg)
```

Comment:
This solution fails if the dataframe has a MultiIndex:

```python
import pandas as pd
import polars as pl

df = pd.DataFrame(
    {
        "a": [1, 2, 3],
        "b": [1, 2, 3],
    },
    index=pd.MultiIndex.from_arrays([(1, 2, 3), (1, 2, 3)], names=["a", "b"])
)
pl.from_pandas(df, include_index=True)
# shape: (3, 2)
# ┌─────┬─────┐
# │ a   ┆ b   │
# │ --- ┆ --- │
# │ i64 ┆ i64 │
# ╞═════╪═════╡
# │ 1   ┆ 1   │
# │ 2   ┆ 2   │
# │ 3   ┆ 3   │
# └─────┴─────┘
```
Also, please add unit tests. I would recommend two separate functions, one similar to the one above that checks for overlap in a MultiIndex, and another that checks with a normal index.